codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -161,17 +178,54 @@ type singlePartition = {
   partitionId: partitionIndex,
 }
 
-let getMostBehindPartition = ({partitions}: t) =>
-  partitions
-  ->List.reduceWithIndex(None, (accum, partition, partitionIndex) => {
-    switch accum {
-    | Some({fetchState: accumPartition}: singlePartition)
-      if accumPartition.FetchState.latestFetchedBlock.blockNumber <
-      partition.latestFetchedBlock.blockNumber => accum
-    | _ => Some({fetchState: partition, partitionId: partitionIndex})
+/**
+Retrieves an array of partitions that are most behind with a max number based on
+the max number of queries with the context of the partitions currently fetching.
+
+The array could be shorter than the max number of queries if the partitions are
+at the max queue size.
+*/
+let getMostBehindPartitions = (
```

Comment:
Read this description to help give context to the function

Reply:
Hmm, this function is definitely a little bit challenging to follow, but starting to get it. At least there are tests :grimacing:

Reply:
Sorry it's probably because of the loop fn. The main reason to do that over just iterating is that you can model an early exit for control flow. Let me add some more comments to it.